src/Type/Constant/ConstantArrayType.php

```diff
@@ -780,6 +781,36 @@ public function unsetOffset(Type $offsetType): Type
 		return new self($this->keyTypes, $this->valueTypes, $this->nextAutoIndexes, $optionalKeys, $isList);
 	}
 
+	public function chunkArray(Type $lengthType, TrinaryLogic $preserveKeys): Type
+	{
+		$biggerOne = IntegerRangeType::fromInterval(1, null);
+		$finiteTypes = $lengthType->getFiniteTypes();
+		if ($biggerOne->isSuperTypeOf($lengthType)->yes() && count($finiteTypes) < self::CHUNK_FINITE_TYPES_LIMIT) {
```

Comment:
maybe?

```suggestion
		$limit = IntegerRangeType::fromInterval(1, self::CHUNK_FINITE_TYPES_LIMIT);
		if ($limit->isSuperTypeOf($lengthType)->yes()) {
  		$finiteTypes = $lengthType->getFiniteTypes();
```

Reply:
that seems to break something. but I just copied it over from the extension, I think you added it originally ;). we could still improve later, I don't have the time and patience to get into this now tbh :)

Reply:
:+1:

Reply:
fyi it was just a off-by-one thing. I adapted the test to check if int<1, 6> is over the limit. I guess it's OK that int<1, 5> is the new allowed upper limit. before it was int<1, 4> because of a `<` check instead of `<=` :)

these range super type checks + finite types are interesting I have to say hmm

Reply:
oh, weird. it breaks another test now, because `int<1, 4>` is considered to be a super type of `60` 🤔

Reply:
ah right, no, not that easy, because e.g. `int<1, 4>` is not a super type of constant int 60, but we still want it to do the high precision chunking. I'll leave it for now :)